TravelAgency-service/src/main/java/cz/muni/fi/travelAgency/service/TripServiceImpl.java

```diff
@@ -40,7 +45,12 @@ public Trip findById(Long id) {
         if (destination == null || destination.isEmpty()){
             throw new IllegalArgumentException("Cannot find Trip with null destination or empty string.");
         }
-        return tripDao.findByDestination(destination);
+        try {
+            return tripDao.findByDestination(destination);
+        } catch (IllegalArgumentException exp){
```

Comment:
This is not the only exception that can be trovn by tha Data acces layer. Chatch all exceptions !!